test/core/tsi/alts/handshaker/alts_handshaker_client_test.cc

```diff
@@ -353,18 +354,33 @@ static void schedule_request_invalid_arg_test() {
   alts_handshaker_client_set_grpc_caller_for_testing(config->client,
                                                      check_must_not_be_called);
   /* Check client_start. */
-  GPR_ASSERT(alts_handshaker_client_start_client(nullptr) ==
-             TSI_INVALID_ARGUMENT);
+  {
+    grpc_core::ExecCtx exec_ctx;
```

Comment:
Is there any reason to make these call sites a part of gRPC core (i.e., by creating `grpc_core::ExecCtx`)?

Reply:
I unfortunately have trouble seeing a good way of doing that, the main thing being that these APIs are normally called from within core, and so can expect to have an `ExecCtx` already available.

A couple of ideas:
1) We could change them to perhaps check if an ExecCtx exists and only instantiate one if so. This would add a decent amount of complication to the code though only for the purpose of this unit test's use case.
2) We could add some wrappers around these, could be exposed with e.g. the rest of the `_test_only` utilities, that could in turn call this APIs but with an ExecCtx

Reply:
Sorry for the confusion. Let me phrase my question differently - what will happen if we do not create an `ExecCtx` before these call sites (e.g., `alts_handshaker_client_start_client`)?

Reply:
In general, the problem this pattern addresses is that we need to make sure that if we're ever scheduling a C-core closure, we need to make sure that we have an `ExecCtx` active for the current thread, otherwise we'll crash.

Before this PR, it just happened that APIs such as `alts_handshaker_client_start_client` didn't schedule any closures internally. However, since this PR starts scheduling the `EnqueueHandshakeAndMaybeStartSomeLocked` call as a closure on a combiner, as a first step of creating a new handshake RPC (from within `alts_handshaker_client_start`), we just need to ensure that we have an ExecCtx active.